src/highlight/highlightUtil.ts

```diff
@@ -94,17 +94,39 @@ export const dedupeRects = (rects: Shape[], threshold = 0.1): Shape[] => {
 };
 
 export const combineRectsByRow = (rects: Shape[]): Shape[] => {
-    const dedupedRects = dedupeRects(rects);
+    const rows: Shape[] = [];
```

Comment:
Nit: maybe we can rename it to combinedRects or shapes? They are not rows anymore.